packages/arb-avm-cpp/avm_values/include/avm_values/tuple.hpp

```diff
@@ -104,61 +102,37 @@ class Tuple {
         }
     }
 
-    void set_element(const uint64_t pos, value newval) {
-        if (pos >= tuple_size()) {
-            throw bad_tuple_index{};
-        }
-        std::shared_ptr<RawTuple> tmp =
-            TuplePool::get_impl().getResource(tuple_size());
-        for (uint64_t i = 0; i < tuple_size(); i++) {
-            if (i == pos) {
-                tmp->data.emplace_back(std::move(newval));
-            } else {
-                tmp->data.emplace_back(tpl->data[i]);
-            }
-        }
-        tpl = std::move(tmp);
-    }
+    void set_element(const uint64_t pos, Value newval);
 
-    [[nodiscard]] value get_element(const uint64_t pos) const {
-        if (pos >= tuple_size()) {
-            throw bad_tuple_index{};
-        }
-        return tpl->data[pos];
-    }
+    [[nodiscard]] Value get_element(const uint64_t pos) const;
```

Comment:
The issue here is that we need to forward declare Value now, because it isn't a std::variant I believe. That said, the forward declarations are a lot simpler now than the previous mess where value was declared before its contents.